Android App with Proguard enabled fails to build in Release mode

### Steps to Reproduce

1. Create new Android App project
2. Set the configuration to Release
3. Select 'Enable Proguard' checkbox in Android project options
4. Save changes
5. Build solution

<!--
If you have a repro project, you may drag & drop the .zip/etc. onto the issue editor to attach it.
-->

### Expected Behavior
Build succeded

### Actual Behavior
Build failed with " "java" exited with code 1." error
### Version Information

<!--
1. On macOS and within Visual Studio, select Visual Studio > About Visual Studio, then click the Show Details button, then click the Copy Information button.

2. Paste below this comment block.
-->
http://xqa.blob.core.windows.net/gist/log-41f6c488213b4fde8589d3c1c517b668.txt

Uses XA Version: 9.0.0.18

### Log File

<!--
1. On macOS and within Visual Studio:
    a. Click **Tools** > **SDK Command Prompt**.
    b. Within the launched `Terminal.app` window, run:

            adb logcat -d | pbcopy

2. Paste below this comment block
-->
Diagnostic build output:
http://xqa.blob.core.windows.net/gist/log-d2228aeedaf348229901105639ceed28.txt

IDE log:
http://xqa.blob.core.windows.net/gist/TestEnableProguard.Ide.log.180726133733-c6efd278b0904f24be5fc76a22cee454.txt

<!--
Switch to the "Preview" tab to ensure your issue renders correctly.

Please add an appropriate "Area:" label in the Labels section.
-->

> VS bug [#655384](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/655384)

I've updated the description to include the relevant log content. In this case, proguard is bailing with a bunch of warnings about unresolved references, as I've pasted below.

I've finally managed to reproduce this with the following steps, and it requires compiling with a `Mono.Android.dll` that is newer than your `android.jar` file.
1) Ensure that API 27 is your latest installed Android platform SDK version.
2) Create a new max compat template. This should result in the following elements being written to your .csproj file:
* `<TargetFrameworkVersion>v9.0</TargetFrameworkVersion>`
* `<AndroidUseLatestPlatformSdk>true</AndroidUseLatestPlatformSdk>`
3) Enable proguard in the release configuration and build.

```
 PROGUARD_HOME=/Library/Frameworks/Xamarin.Android.framework/Libraries/xbuild/Xamarin/Android/proguard
    /usr/bin/java -jar /Library/Frameworks/Xamarin.Android.framework/Libraries/xbuild/Xamarin/Android/proguard/lib/proguard.jar -include '/Users/builder/Library/Developer/Xamarin/android-sdk-macosx//tools/proguard/proguard-android.txt' -include 'obj/Release/proguard/proguard_xamarin.cfg' -include 'obj/Release/proguard/proguard_project_references.cfg' -include 'obj/Release/proguard/proguard_project_primary.cfg' -include '/Users/builder/agent/_work/r27/a/XQA/XQA.VisualStudioMac/TestResults/61c53ceb/TestEnableProguard/Temp/DroidApp/packages/Xamarin.Android.Arch.Lifecycle.Runtime.1.0.3/build/MonoAndroid80/../../proguard/proguard.txt' -include '/Users/builder/agent/_work/r27/a/XQA/XQA.VisualStudioMac/TestResults/61c53ceb/TestEnableProguard/Temp/DroidApp/packages/Xamarin.Android.Support.Core.UI.27.0.2/build/MonoAndroid81/../../proguard/proguard.txt' -include '/Users/builder/agent/_work/r27/a/XQA/XQA.VisualStudioMac/TestResults/61c53ceb/TestEnableProguard/Temp/DroidApp/packages/Xamarin.Android.Support.Media.Compat.27.0.2/build/MonoAndroid81/../../proguard/proguard.txt' -include '/Users/builder/agent/_work/r27/a/XQA/XQA.VisualStudioMac/TestResults/61c53ceb/TestEnableProguard/Temp/DroidApp/packages/Xamarin.Android.Support.Animated.Vector.Drawable.27.0.2/build/MonoAndroid81/../../proguard/proguard.txt' -injars "'obj/Release/android/bin/classes/../classes.zip'(!META-INF/MANIFEST.MF):'/Library/Frameworks/Mono.framework/External/xbuild-frameworks/MonoAndroid/v9.0/mono.android.jar'(!META-INF/MANIFEST.MF):'obj/Release/lp/1/jl/arch-core-common.jar'(!META-INF/MANIFEST.MF):'obj/Release/lp/10/jl/bin/classes.jar'(!META-INF/MANIFEST.MF):'obj/Release/lp/11/jl/bin/classes.jar'(!META-INF/MANIFEST.MF):'obj/Release/lp/12/jl/bin/classes.jar'(!META-INF/MANIFEST.MF):'obj/Release/lp/2/jl/arch-lifecycle-common.jar'(!META-INF/MANIFEST.MF):'obj/Release/lp/3/jl/bin/classes.jar'(!META-INF/MANIFEST.MF):'obj/Release/lp/4/jl/bin/classes.jar'(!META-INF/MANIFEST.MF):'obj/Release/lp/6/jl/bin/classes.jar'(!META-INF/MANIFEST.MF):'obj/Release/lp/7/jl/bin/classes.jar'(!META-INF/MANIFEST.MF):'obj/Release/lp/8/jl/bin/classes.jar'(!META-INF/MANIFEST.MF):'obj/Release/lp/9/jl/bin/classes.jar'(!META-INF/MANIFEST.MF)" -libraryjars '/Users/builder/Library/Developer/Xamarin/android-sdk-macosx/platforms/android-27/android.jar' -outjars obj/Release/proguard/__proguard_output__.jar -optimizations !code/allocation/variable 
    ProGuard, version 5.3.2
    Note: the configuration doesn't specify which class members to keep for class 'android.arch.core.internal.SafeIterableMap$SupportRemove'
    Note: there were 1 '-keepclassmembers' options that didn't specify class
          members. You should specify at least some class members or consider
          if you just need '-keep'.
          (http://proguard.sourceforge.net/manual/troubleshooting.html#classmembers)
    Reading input...
    Reading program zip [/Users/builder/agent/_work/r27/a/XQA/XQA.VisualStudioMac/TestResults/61c53ceb/TestEnableProguard/Temp/DroidApp/DroidApp/obj/Release/android/bin/classes.zip] (filtered)
    Reading program jar [/Library/Frameworks/Xamarin.Android.framework/Versions/9.0.0-18/lib/xamarin.android/xbuild-frameworks/MonoAndroid/v9.0/mono.android.jar] (filtered)
    Reading program jar [/Users/builder/agent/_work/r27/a/XQA/XQA.VisualStudioMac/TestResults/61c53ceb/TestEnableProguard/Temp/DroidApp/DroidApp/obj/Release/lp/1/jl/arch-core-common.jar] (filtered)
    Reading program jar [/Users/builder/agent/_work/r27/a/XQA/XQA.VisualStudioMac/TestResults/61c53ceb/TestEnableProguard/Temp/DroidApp/DroidApp/obj/Release/lp/10/jl/bin/classes.jar] (filtered)
    Reading program jar [/Users/builder/agent/_work/r27/a/XQA/XQA.VisualStudioMac/TestResults/61c53ceb/TestEnableProguard/Temp/DroidApp/DroidApp/obj/Release/lp/11/jl/bin/classes.jar] (filtered)
    Reading program jar [/Users/builder/agent/_work/r27/a/XQA/XQA.VisualStudioMac/TestResults/61c53ceb/TestEnableProguard/Temp/DroidApp/DroidApp/obj/Release/lp/12/jl/bin/classes.jar] (filtered)
    Reading program jar [/Users/builder/agent/_work/r27/a/XQA/XQA.VisualStudioMac/TestResults/61c53ceb/TestEnableProguard/Temp/DroidApp/DroidApp/obj/Release/lp/2/jl/arch-lifecycle-common.jar] (filtered)
    Reading program jar [/Users/builder/agent/_work/r27/a/XQA/XQA.VisualStudioMac/TestResults/61c53ceb/TestEnableProguard/Temp/DroidApp/DroidApp/obj/Release/lp/3/jl/bin/classes.jar] (filtered)
    Reading program jar [/Users/builder/agent/_work/r27/a/XQA/XQA.VisualStudioMac/TestResults/61c53ceb/TestEnableProguard/Temp/DroidApp/DroidApp/obj/Release/lp/4/jl/bin/classes.jar] (filtered)
    Reading program jar [/Users/builder/agent/_work/r27/a/XQA/XQA.VisualStudioMac/TestResults/61c53ceb/TestEnableProguard/Temp/DroidApp/DroidApp/obj/Release/lp/6/jl/bin/classes.jar] (filtered)
    Reading program jar [/Users/builder/agent/_work/r27/a/XQA/XQA.VisualStudioMac/TestResults/61c53ceb/TestEnableProguard/Temp/DroidApp/DroidApp/obj/Release/lp/7/jl/bin/classes.jar] (filtered)
    Reading program jar [/Users/builder/agent/_work/r27/a/XQA/XQA.VisualStudioMac/TestResults/61c53ceb/TestEnableProguard/Temp/DroidApp/DroidApp/obj/Release/lp/8/jl/bin/classes.jar] (filtered)
    Reading program jar [/Users/builder/agent/_work/r27/a/XQA/XQA.VisualStudioMac/TestResults/61c53ceb/TestEnableProguard/Temp/DroidApp/DroidApp/obj/Release/lp/9/jl/bin/classes.jar] (filtered)
    Reading library jar [/Users/builder/Library/Developer/Xamarin/android-sdk-macosx/platforms/android-27/android.jar]
    Initializing...
    Note: the configuration refers to the unknown class 'com.google.vending.licensing.ILicensingService'
    Note: the configuration refers to the unknown class 'com.android.vending.licensing.ILicensingService'
    Note: the configuration refers to the unknown class 'android.support.annotation.Keep'
    Note: the configuration refers to the unknown class 'android.support.annotation.Keep'
    Note: the configuration refers to the unknown class 'android.support.annotation.Keep'
    Note: the configuration refers to the unknown class 'android.support.annotation.Keep'
    Note: the configuration refers to the unknown class 'android.support.annotation.Keep'
    Note: android.support.v4.app.NotificationCompatJellybean calls 'Field.getType'
    PROGUARD : warning : mono.android.app.admin.DevicePolicyManager_OnClearApplicationUserDataListenerImplementor: can't find superclass or interface android.app.admin.DevicePolicyManager$OnClearApplicationUserDataListener
    PROGUARD : warning : mono.android.graphics.ImageDecoder_OnHeaderDecodedListenerImplementor: can't find superclass or interface android.graphics.ImageDecoder$OnHeaderDecodedListener
    PROGUARD : warning : mono.android.graphics.ImageDecoder_OnPartialImageListenerImplementor: can't find superclass or interface android.graphics.ImageDecoder$OnPartialImageListener
    PROGUARD : warning : mono.android.media.MediaPlayer_OnMediaTimeDiscontinuityListenerImplementor: can't find superclass or interface android.media.MediaPlayer$OnMediaTimeDiscontinuityListener
    PROGUARD : warning : mono.android.media.MediaPlayer_OnSubtitleDataListenerImplementor: can't find superclass or interface android.media.MediaPlayer$OnSubtitleDataListener
    PROGUARD : warning : mono.android.os.StrictMode_OnThreadViolationListenerImplementor: can't find superclass or interface android.os.StrictMode$OnThreadViolationListener
    PROGUARD : warning : mono.android.os.StrictMode_OnVmViolationListenerImplementor: can't find superclass or interface android.os.StrictMode$OnVmViolationListener
    PROGUARD : warning : mono.android.se.omapi.SEService_OnConnectedListenerImplementor: can't find superclass or interface android.se.omapi.SEService$OnConnectedListener
    PROGUARD : warning : mono.android.view.View_OnUnhandledKeyEventListenerImplementor: can't find superclass or interface android.view.View$OnUnhandledKeyEventListener
    PROGUARD : warning : mono.android.app.admin.DevicePolicyManager_OnClearApplicationUserDataListenerImplementor: can't find referenced class android.app.admin.DevicePolicyManager$OnClearApplicationUserDataListener
    PROGUARD : warning : mono.android.graphics.ImageDecoder_OnHeaderDecodedListenerImplementor: can't find referenced class android.graphics.ImageDecoder$OnHeaderDecodedListener
    PROGUARD : warning : mono.android.graphics.ImageDecoder_OnHeaderDecodedListenerImplementor: can't find referenced class android.graphics.ImageDecoder$ImageInfo
    PROGUARD : warning : mono.android.graphics.ImageDecoder_OnHeaderDecodedListenerImplementor: can't find referenced class android.graphics.ImageDecoder$Source
    PROGUARD : warning : mono.android.graphics.ImageDecoder_OnHeaderDecodedListenerImplementor: can't find referenced class android.graphics.ImageDecoder
    PROGUARD : warning : mono.android.graphics.ImageDecoder_OnHeaderDecodedListenerImplementor: can't find referenced class android.graphics.ImageDecoder
    PROGUARD : warning : mono.android.graphics.ImageDecoder_OnHeaderDecodedListenerImplementor: can't find referenced class android.graphics.ImageDecoder$ImageInfo
    PROGUARD : warning : mono.android.graphics.ImageDecoder_OnHeaderDecodedListenerImplementor: can't find referenced class android.graphics.ImageDecoder$Source
    PROGUARD : warning : mono.android.graphics.ImageDecoder_OnHeaderDecodedListenerImplementor: can't find referenced class android.graphics.ImageDecoder
    PROGUARD : warning : mono.android.graphics.ImageDecoder_OnHeaderDecodedListenerImplementor: can't find referenced class android.graphics.ImageDecoder$ImageInfo
    PROGUARD : warning : mono.android.graphics.ImageDecoder_OnHeaderDecodedListenerImplementor: can't find referenced class android.graphics.ImageDecoder$Source
    PROGUARD : warning : mono.android.graphics.ImageDecoder_OnPartialImageListenerImplementor: can't find referenced class android.graphics.ImageDecoder$OnPartialImageListener
    PROGUARD : warning : mono.android.graphics.ImageDecoder_OnPartialImageListenerImplementor: can't find referenced class android.graphics.ImageDecoder$DecodeException
    PROGUARD : warning : mono.android.graphics.ImageDecoder_OnPartialImageListenerImplementor: can't find referenced class android.graphics.ImageDecoder
    PROGUARD : warning : mono.android.graphics.ImageDecoder_OnPartialImageListenerImplementor: can't find referenced class android.graphics.ImageDecoder$DecodeException
    PROGUARD : warning : mono.android.graphics.ImageDecoder_OnPartialImageListenerImplementor: can't find referenced class android.graphics.ImageDecoder$DecodeException
    PROGUARD : warning : mono.android.media.MediaPlayer_OnMediaTimeDiscontinuityListenerImplementor: can't find referenced class android.media.MediaPlayer$OnMediaTimeDiscontinuityListener
    PROGUARD : warning : mono.android.media.MediaPlayer_OnSubtitleDataListenerImplementor: can't find referenced class android.media.MediaPlayer$OnSubtitleDataListener
    PROGUARD : warning : mono.android.media.MediaPlayer_OnSubtitleDataListenerImplementor: can't find referenced class android.media.SubtitleData
    PROGUARD : warning : mono.android.media.MediaPlayer_OnSubtitleDataListenerImplementor: can't find referenced class android.media.SubtitleData
    PROGUARD : warning : mono.android.os.StrictMode_OnThreadViolationListenerImplementor: can't find referenced class android.os.StrictMode$OnThreadViolationListener
    PROGUARD : warning : mono.android.os.StrictMode_OnThreadViolationListenerImplementor: can't find referenced class android.os.strictmode.Violation
    PROGUARD : warning : mono.android.os.StrictMode_OnThreadViolationListenerImplementor: can't find referenced class android.os.strictmode.Violation
    PROGUARD : warning : mono.android.os.StrictMode_OnVmViolationListenerImplementor: can't find referenced class android.os.StrictMode$OnVmViolationListener
    PROGUARD : warning : mono.android.os.StrictMode_OnVmViolationListenerImplementor: can't find referenced class android.os.strictmode.Violation
    PROGUARD : warning : mono.android.os.StrictMode_OnVmViolationListenerImplementor: can't find referenced class android.os.strictmode.Violation
    PROGUARD : warning : mono.android.se.omapi.SEService_OnConnectedListenerImplementor: can't find referenced class android.se.omapi.SEService$OnConnectedListener
    PROGUARD : warning : mono.android.se.omapi.SEService_OnConnectedListenerImplementor: can't find referenced class android.se.omapi.SEService
    PROGUARD : warning : mono.android.view.View_OnUnhandledKeyEventListenerImplementor: can't find referenced class android.view.View$OnUnhandledKeyEventListener
    Note: android.support.v4.app.FragmentTransition: can't find dynamically referenced class android.support.transition.FragmentTransitionSupport
    Note: android.support.v4.graphics.TypefaceCompatApi24Impl: can't find dynamically referenced class android.graphics.FontFamily
    Note: android.support.v4.graphics.TypefaceCompatApi26Impl: can't find dynamically referenced class android.graphics.FontFamily
    Note: android.support.v4.media.ParceledListSliceAdapterApi21: can't find dynamically referenced class android.content.pm.ParceledListSlice
    Note: android.support.v4.text.ICUCompat: can't find dynamically referenced class libcore.icu.ICU
    Note: android.support.v4.text.ICUCompat: can't find dynamically referenced class libcore.icu.ICU
    Note: android.support.v7.app.ResourcesFlusher: can't find dynamically referenced class android.content.res.ThemedResourceCache
    Note: android.support.v7.widget.DrawableUtils: can't find dynamically referenced class android.graphics.Insets
    Note: the configuration keeps the entry point 'android.support.graphics.drawable.VectorDrawableCompat$VPath { void setPathData(android.support.v4.graphics.PathParser$PathDataNode[]); }', but not the descriptor class 'android.support.v4.graphics.PathParser$PathDataNode'
    Note: the configuration keeps the entry point 'android.support.v4.view.ViewPager { void setAdapter(android.support.v4.view.PagerAdapter); }', but not the descriptor class 'android.support.v4.view.PagerAdapter'
    Note: the configuration keeps the entry point 'android.support.v4.view.ViewPager { void setOnPageChangeListener(android.support.v4.view.ViewPager$OnPageChangeListener); }', but not the descriptor class 'android.support.v4.view.ViewPager$OnPageChangeListener'
    Note: the configuration keeps the entry point 'android.support.v4.view.ViewPager { android.support.v4.view.ViewPager$OnPageChangeListener setInternalPageChangeListener(android.support.v4.view.ViewPager$OnPageChangeListener); }', but not the descriptor class 'android.support.v4.view.ViewPager$OnPageChangeListener'
    Note: the configuration keeps the entry point 'android.support.v4.widget.DrawerLayout { void setDrawerListener(android.support.v4.widget.DrawerLayout$DrawerListener); }', but not the descriptor class 'android.support.v4.widget.DrawerLayout$DrawerListener'
    Note: the configuration keeps the entry point 'android.support.v4.widget.NestedScrollView { void setOnScrollChangeListener(android.support.v4.widget.NestedScrollView$OnScrollChangeListener); }', but not the descriptor class 'android.support.v4.widget.NestedScrollView$OnScrollChangeListener'
    Note: the configuration keeps the entry point 'android.support.v4.widget.SlidingPaneLayout { void setPanelSlideListener(android.support.v4.widget.SlidingPaneLayout$PanelSlideListener); }', but not the descriptor class 'android.support.v4.widget.SlidingPaneLayout$PanelSlideListener'
    Note: the configuration keeps the entry point 'android.support.v4.widget.SwipeRefreshLayout { void setOnRefreshListener(android.support.v4.widget.SwipeRefreshLayout$OnRefreshListener); }', but not the descriptor class 'android.support.v4.widget.SwipeRefreshLayout$OnRefreshListener'
    Note: the configuration keeps the entry point 'android.support.v4.widget.SwipeRefreshLayout { void setOnChildScrollUpCallback(android.support.v4.widget.SwipeRefreshLayout$OnChildScrollUpCallback); }', but not the descriptor class 'android.support.v4.widget.SwipeRefreshLayout$OnChildScrollUpCallback'
    Note: the configuration keeps the entry point 'android.support.v7.view.menu.ActionMenuItemView { void setItemInvoker(android.support.v7.view.menu.MenuBuilder$ItemInvoker); }', but not the descriptor class 'android.support.v7.view.menu.MenuBuilder$ItemInvoker'
    Note: the configuration keeps the entry point 'android.support.v7.view.menu.ActionMenuItemView { void setPopupCallback(android.support.v7.view.menu.ActionMenuItemView$PopupCallback); }', but not the descriptor class 'android.support.v7.view.menu.ActionMenuItemView$PopupCallback'
    Note: the configuration keeps the entry point 'android.support.v7.widget.ActionBarContainer { void setTabContainer(android.support.v7.widget.ScrollingTabContainerView); }', but not the descriptor class 'android.support.v7.widget.ScrollingTabContainerView'
    Note: the configuration keeps the entry point 'android.support.v7.widget.ActionBarOverlayLayout { void setActionBarVisibilityCallback(android.support.v7.widget.ActionBarOverlayLayout$ActionBarVisibilityCallback); }', but not the descriptor class 'android.support.v7.widget.ActionBarOverlayLayout$ActionBarVisibilityCallback'
    Note: the configuration keeps the entry point 'android.support.v7.widget.ActionMenuView { void setPresenter(android.support.v7.widget.ActionMenuPresenter); }', but not the descriptor class 'android.support.v7.widget.ActionMenuPresenter'
    Note: the configuration keeps the entry point 'android.support.v7.widget.ActionMenuView { void setOnMenuItemClickListener(android.support.v7.widget.ActionMenuView$OnMenuItemClickListener); }', but not the descriptor class 'android.support.v7.widget.ActionMenuView$OnMenuItemClickListener'
    Note: the configuration keeps the entry point 'android.support.v7.widget.ActivityChooserView { void setActivityChooserModel(android.support.v7.widget.ActivityChooserModel); }', but not the descriptor class 'android.support.v7.widget.ActivityChooserModel'
    Note: the configuration keeps the entry point 'android.support.v7.widget.ActivityChooserView { void setProvider(android.support.v4.view.ActionProvider); }', but not the descriptor class 'android.support.v4.view.ActionProvider'
    Note: the configuration keeps the entry point 'android.support.v7.widget.ContentFrameLayout { void setAttachListener(android.support.v7.widget.ContentFrameLayout$OnAttachListener); }', but not the descriptor class 'android.support.v7.widget.ContentFrameLayout$OnAttachListener'
    Note: the configuration keeps the entry point 'android.support.v7.widget.FitWindowsFrameLayout { void setOnFitSystemWindowsListener(android.support.v7.widget.FitWindowsViewGroup$OnFitSystemWindowsListener); }', but not the descriptor class 'android.support.v7.widget.FitWindowsViewGroup$OnFitSystemWindowsListener'
    Note: the configuration keeps the entry point 'android.support.v7.widget.FitWindowsLinearLayout { void setOnFitSystemWindowsListener(android.support.v7.widget.FitWindowsViewGroup$OnFitSystemWindowsListener); }', but not the descriptor class 'android.support.v7.widget.FitWindowsViewGroup$OnFitSystemWindowsListener'
    Note: the configuration keeps the entry point 'android.support.v7.widget.MenuPopupWindow$MenuDropDownListView { void setHoverListener(android.support.v7.widget.MenuItemHoverListener); }', but not the descriptor class 'android.support.v7.widget.MenuItemHoverListener'
    Note: the configuration keeps the entry point 'android.support.v7.widget.SearchView { void setOnQueryTextListener(android.support.v7.widget.SearchView$OnQueryTextListener); }', but not the descriptor class 'android.support.v7.widget.SearchView$OnQueryTextListener'
    Note: the configuration keeps the entry point 'android.support.v7.widget.SearchView { void setOnCloseListener(android.support.v7.widget.SearchView$OnCloseListener); }', but not the descriptor class 'android.support.v7.widget.SearchView$OnCloseListener'
    Note: the configuration keeps the entry point 'android.support.v7.widget.SearchView { void setOnSuggestionListener(android.support.v7.widget.SearchView$OnSuggestionListener); }', but not the descriptor class 'android.support.v7.widget.SearchView$OnSuggestionListener'
    Note: the configuration keeps the entry point 'android.support.v7.widget.SearchView { void setSuggestionsAdapter(android.support.v4.widget.CursorAdapter); }', but not the descriptor class 'android.support.v4.widget.CursorAdapter'
    Note: the configuration keeps the entry point 'android.support.v7.widget.SearchView$SearchAutoComplete { void setSearchView(android.support.v7.widget.SearchView); }', but not the descriptor class 'android.support.v7.widget.SearchView'
    Note: the configuration keeps the entry point 'android.support.v7.widget.Toolbar { void setOnMenuItemClickListener(android.support.v7.widget.Toolbar$OnMenuItemClickListener); }', but not the descriptor class 'android.support.v7.widget.Toolbar$OnMenuItemClickListener'
    Note: the configuration keeps the entry point 'android.support.v7.widget.ViewStubCompat { void setOnInflateListener(android.support.v7.widget.ViewStubCompat$OnInflateListener); }', but not the descriptor class 'android.support.v7.widget.ViewStubCompat$OnInflateListener'
    Note: there were 7 references to unknown classes.
          You should check your configuration for typos.
          (http://proguard.sourceforge.net/manual/troubleshooting.html#unknownclass)
    Note: there were 1 classes trying to access generic signatures using reflection.
    PROGUARD : warning : there were 38 unresolved references to classes or interfaces.
             You may need to add missing library jars or update their versions.
             If your code works fine without the missing classes, you can suppress
             the warnings with '-dontwarn' options.
             (http://proguard.sourceforge.net/manual/troubleshooting.html#unresolvedclass)
    java.io.IOException: Please correct the above warnings first.
    	at proguard.Initializer.execute(Unknown Source)
    	at proguard.ProGuard.initialize(Unknown Source)
    	at proguard.ProGuard.execute(Unknown Source)
    	at proguard.ProGuard.main(Unknown Source)
          You should consider keeping the signature attributes
          (using '-keepattributes Signature').
          (http://proguard.sourceforge.net/manual/troubleshooting.html#attributes)
    Note: there were 28 unkept descriptor classes in kept class members.
          You should consider explicitly keeping the mentioned classes
          (using '-keep').
          (http://proguard.sourceforge.net/manual/troubleshooting.html#descriptorclass)
    Note: there were 8 unresolved dynamic references to classes or interfaces.
          You should check if you need to specify additional program jars.
          (http://proguard.sourceforge.net/manual/troubleshooting.html#dynamicalclass)
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(2465,3): error MSB6006: "java" exited with code 1.
  Done executing task "Proguard" -- FAILED.
```

--- a/Documentation/guides/BuildProcess.md
+++ b/Documentation/guides/BuildProcess.md
@@ -624,7 +624,7 @@ when packaing Release applications.
     allows the developer to define custom items to use with the
     `AndroidVersionCodePattern`. They are in the form of a `key=value`
     pair. All items in the `value` should be integer values. For
-    example: `screen=23;target=$(_SupportedApiLevel)`. As you can see
+    example: `screen=23;target=$(_AndroidApiLevel)`. As you can see
     you can make use of existing or custom MSBuild properties in the
     string.
 

--- a/src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs
+++ b/src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs
@@ -294,10 +294,10 @@ namespace Xamarin.Android.Tasks
 
 				for (int apiLevel = maxSupported; apiLevel >= MonoAndroidHelper.SupportedVersions.MinStableVersion.ApiLevel; apiLevel--) {
 					var id = MonoAndroidHelper.SupportedVersions.GetIdFromApiLevel (apiLevel);
-					var apiPlatformDir = Path.Combine (AndroidSdkPath, "platforms", "android-" + id);
-					if (Directory.Exists (apiPlatformDir)) {
+					var apiPlatformDir = MonoAndroidHelper.AndroidSdk.TryGetPlatformDirectoryFromApiLevel (id, MonoAndroidHelper.SupportedVersions);
+					if (apiPlatformDir != null && Directory.Exists (apiPlatformDir)) {
 						var targetFramework = MonoAndroidHelper.SupportedVersions.GetFrameworkVersionFromId (id);
-						if (targetFramework != null && MonoAndroidHelper.SupportedVersions.FrameworkDirectories.Any (p => Directory.Exists (Path.Combine (p, targetFramework)))) {
+						if (targetFramework != null && MonoAndroidHelper.SupportedVersions.InstalledBindingVersions.Any (b => b.FrameworkVersion == targetFramework)) {
 							AndroidApiLevel = apiLevel.ToString ();
 							TargetFrameworkVersion = targetFramework;
 							break;

--- a/src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs
+++ b/src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs
@@ -272,24 +272,38 @@ namespace Xamarin.Android.Tasks
 				(string.IsNullOrWhiteSpace (AndroidApiLevel) && string.IsNullOrWhiteSpace (TargetFrameworkVersion));
 
 			if (UseLatestAndroidPlatformSdk) {
-				AndroidApiLevel = GetMaxInstalledApiLevel ().ToString ();
-				SupportedApiLevel = GetMaxStableApiLevel ().ToString ();
-				int maxInstalled, maxSupported = 0;
-				if (int.TryParse (AndroidApiLevel, out maxInstalled) && int.TryParse (SupportedApiLevel, out maxSupported) && maxInstalled > maxSupported) {
+				int maxInstalled = GetMaxInstalledApiLevel ();
+				int maxSupported = GetMaxStableApiLevel ();
+				AndroidApiLevel = maxInstalled.ToString ();
+				SupportedApiLevel = maxSupported.ToString ();
+				if (maxInstalled > maxSupported) {
 					Log.LogDebugMessage ($"API Level {AndroidApiLevel} is greater than the maximum supported API level of {SupportedApiLevel}. " +
 						"Support for this API will be added in a future release.");
-					AndroidApiLevel = SupportedApiLevel;
 				}
 				if (!string.IsNullOrWhiteSpace (TargetFrameworkVersion)) {
 					var userSelected = MonoAndroidHelper.SupportedVersions.GetApiLevelFromFrameworkVersion (TargetFrameworkVersion);
 					// overwrite using user version only if it is 
 					// above the maxStableApi and a valid apiLevel.
 					if (userSelected != null && userSelected > maxSupported && userSelected <= maxInstalled) {
-						AndroidApiLevel = userSelected.ToString ();
-						SupportedApiLevel = userSelected.ToString ();
+						maxInstalled =
+							maxSupported = userSelected.Value;
+						AndroidApiLevel =
+							SupportedApiLevel = userSelected.ToString ();
 					}
 				}
-				TargetFrameworkVersion = GetTargetFrameworkVersionFromApiLevel ();
+
+				for (int apiLevel = maxSupported; apiLevel >= MonoAndroidHelper.SupportedVersions.MinStableVersion.ApiLevel; apiLevel--) {
+					var id = MonoAndroidHelper.SupportedVersions.GetIdFromApiLevel (apiLevel);
+					var apiPlatformDir = Path.Combine (AndroidSdkPath, "platforms", "android-" + id);
+					if (Directory.Exists (apiPlatformDir)) {
+						var targetFramework = MonoAndroidHelper.SupportedVersions.GetFrameworkVersionFromId (id);
+						if (targetFramework != null && MonoAndroidHelper.SupportedVersions.FrameworkDirectories.Any (p => Directory.Exists (Path.Combine (p, targetFramework)))) {
+							AndroidApiLevel = apiLevel.ToString ();
+							TargetFrameworkVersion = targetFramework;
+							break;
+						}
+					}
+				}
 				return TargetFrameworkVersion != null;
 			}
 

--- a/src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs
+++ b/src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs
@@ -41,9 +41,6 @@ namespace Xamarin.Android.Tasks
 		public string AndroidApiLevelName { get; set; }
 
 		[Output]
-		public string SupportedApiLevel { get; set; }
-
-		[Output]
 		public string AndroidSdkBuildToolsPath { get; set; }
 
 		[Output]
@@ -208,7 +205,6 @@ namespace Xamarin.Android.Tasks
 			Log.LogDebugMessage ($"  {nameof (TargetFrameworkVersion)}: {TargetFrameworkVersion}");
 			Log.LogDebugMessage ($"  {nameof (AndroidApiLevel)}: {AndroidApiLevel}");
 			Log.LogDebugMessage ($"  {nameof (AndroidApiLevelName)}: {AndroidApiLevelName}");
-			Log.LogDebugMessage ($"  {nameof (SupportedApiLevel)}: {SupportedApiLevel}");
 			Log.LogDebugMessage ($"  {nameof (AndroidSdkBuildToolsPath)}: {AndroidSdkBuildToolsPath}");
 			Log.LogDebugMessage ($"  {nameof (AndroidSdkBuildToolsBinPath)}: {AndroidSdkBuildToolsBinPath}");
 			Log.LogDebugMessage ($"  {nameof (ZipAlignPath)}: {ZipAlignPath}");
@@ -275,9 +271,8 @@ namespace Xamarin.Android.Tasks
 				int maxInstalled = GetMaxInstalledApiLevel ();
 				int maxSupported = GetMaxStableApiLevel ();
 				AndroidApiLevel = maxInstalled.ToString ();
-				SupportedApiLevel = maxSupported.ToString ();
 				if (maxInstalled > maxSupported) {
-					Log.LogDebugMessage ($"API Level {AndroidApiLevel} is greater than the maximum supported API level of {SupportedApiLevel}. " +
+					Log.LogDebugMessage ($"API Level {maxInstalled} is greater than the maximum supported API level of {maxSupported}. " +
 						"Support for this API will be added in a future release.");
 				}
 				if (!string.IsNullOrWhiteSpace (TargetFrameworkVersion)) {
@@ -287,8 +282,7 @@ namespace Xamarin.Android.Tasks
 					if (userSelected != null && userSelected > maxSupported && userSelected <= maxInstalled) {
 						maxInstalled =
 							maxSupported = userSelected.Value;
-						AndroidApiLevel =
-							SupportedApiLevel = userSelected.ToString ();
+						AndroidApiLevel = userSelected.ToString ();
 					}
 				}
 
@@ -317,13 +311,11 @@ namespace Xamarin.Android.Tasks
 					return false;
 				}
 				AndroidApiLevel = MonoAndroidHelper.SupportedVersions.GetApiLevelFromId (id).ToString ();
-				SupportedApiLevel = AndroidApiLevel;
 				return true;
 			}
 
 			if (!string.IsNullOrWhiteSpace (AndroidApiLevel)) {
 				AndroidApiLevel = AndroidApiLevel.Trim ();
-				SupportedApiLevel = GetMaxSupportedApiLevel (AndroidApiLevel);
 				TargetFrameworkVersion = GetTargetFrameworkVersionFromApiLevel ();
 				return TargetFrameworkVersion != null;
 			}
@@ -383,8 +375,7 @@ namespace Xamarin.Android.Tasks
 
 		string GetTargetFrameworkVersionFromApiLevel ()
 		{
-			string targetFramework = MonoAndroidHelper.SupportedVersions.GetFrameworkVersionFromId (SupportedApiLevel) ??
-				MonoAndroidHelper.SupportedVersions.GetFrameworkVersionFromId (AndroidApiLevel);
+			string targetFramework = MonoAndroidHelper.SupportedVersions.GetFrameworkVersionFromId (AndroidApiLevel);
 			if (targetFramework != null)
 				return targetFramework;
 			Log.LogCodedError ("XA0000",

--- a/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/ResolveSdksTaskTests.cs
+++ b/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/ResolveSdksTaskTests.cs
@@ -190,7 +190,7 @@ namespace Xamarin.Android.Build.Tests {
 				SequencePointsMode = "None",
 			};
 			Assert.AreEqual (expectedTaskResult, resolveSdks.Execute () && validateJavaVersion.Execute () && androidTooling.Execute (), $"Tasks should have {(expectedTaskResult ? "succeeded" : "failed" )}.");
-			Assert.AreEqual (expectedTargetFramework, androidTooling.TargetFrameworkVersion, $"TargetFrameworkVersion should be {expectedTargetFramework} but was {targetFrameworkVersion}");
+			Assert.AreEqual (expectedTargetFramework, androidTooling.TargetFrameworkVersion, $"TargetFrameworkVersion should be {expectedTargetFramework} but was {androidTooling.TargetFrameworkVersion}");
 			if (!string.IsNullOrWhiteSpace (expectedError)) {
 				Assert.AreEqual (1, errors.Count (), "An error should have been raised.");
 				Assert.AreEqual (expectedError, errors [0].Code, $"Expected error code {expectedError} but found {errors [0].Code}");
@@ -240,7 +240,7 @@ namespace Xamarin.Android.Build.Tests {
 				UseLatestAndroidPlatformSdk = false,
 				AotAssemblies = false,
 				SequencePointsMode = "None",
-			};;
+			};
 			var start = DateTime.UtcNow;
 			Assert.IsTrue (resolveSdks.Execute (), "ResolveSdks should succeed!");
 			Assert.IsTrue (validateJavaVersion.Execute (), "ValidateJavaVersion should succeed!");
@@ -276,6 +276,142 @@ namespace Xamarin.Android.Build.Tests {
 			Assert.AreEqual (androidTooling.AndroidUseApkSigner, false, "AndroidUseApkSigner should be false");
 			Assert.AreEqual (validateJavaVersion.JdkVersion, "1.8.0", "JdkVersion should be 1.8.0");
 			Assert.AreEqual (validateJavaVersion.MinimumRequiredJdkVersion, "1.8", "MinimumRequiredJdkVersion should be 1.8");
+			Directory.Delete (Path.Combine (Root, path), recursive: true);
+		}
+
+		static object [] TargetFrameworkPairingParameters = new [] {
+			//We support 28, but only 27 is installed
+			new object [] {
+				"Older API Installed", //description
+				// androidSdks
+				new [] {
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+				},
+				// targetFrameworks
+				new ApiInfo [] {
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+					new ApiInfo () { Id = "28", Level = 28, Name = "P",    FrameworkVersion = "v9.0", Stable = true },
+				},
+				null,   //userSelected
+				"27",   //androidApiLevel
+				"27",   //androidApiLevelName
+				"v8.1", //targetFrameworkVersion
+				"28",   //supportedApiLevel
+			},
+			//28 is installed but we only support 27
+			new object [] {
+				"Newer API Installed", //description
+				// androidSdks
+				new [] {
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+					new ApiInfo () { Id = "28", Level = 28, Name = "P",    FrameworkVersion = "v9.0", Stable = true },
+				},
+				// targetFrameworks
+				new ApiInfo [] {
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+				},
+				null,   //userSelected
+				"27",   //androidApiLevel
+				"27",   //androidApiLevelName
+				"v8.1", //targetFrameworkVersion
+				"27",   //supportedApiLevel
+			},
+			//A paired downgrade to API 26
+			new object [] {
+				"Paired Downgrade", //description
+				// androidSdks
+				new [] {
+					new ApiInfo () { Id = "26", Level = 26, Name = "Oreo", FrameworkVersion = "v8.0", Stable = true },
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+				},
+				// targetFrameworks
+				new ApiInfo [] {
+					new ApiInfo () { Id = "26", Level = 26, Name = "Oreo", FrameworkVersion = "v8.0", Stable = true },
+					new ApiInfo () { Id = "28", Level = 28, Name = "P",    FrameworkVersion = "v9.0", Stable = true },
+				},
+				null,   //userSelected
+				"26",   //androidApiLevel
+				"26",   //androidApiLevelName
+				"v8.0", //targetFrameworkVersion
+				"28",   //supportedApiLevel
+			},
+			//A new API level 28 is not stable yet
+			new object [] {
+				"New Unstable API", //description
+				// androidSdks
+				new [] {
+					new ApiInfo () { Id = "26", Level = 26, Name = "Oreo", FrameworkVersion = "v8.0", Stable = true },
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+					new ApiInfo () { Id = "28", Level = 28, Name = "P",    FrameworkVersion = "v9.0", Stable = false },
+				},
+				// targetFrameworks
+				new ApiInfo [] {
+					new ApiInfo () { Id = "26", Level = 26, Name = "Oreo", FrameworkVersion = "v8.0", Stable = true },
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+					new ApiInfo () { Id = "28", Level = 28, Name = "P",    FrameworkVersion = "v9.0", Stable = false },
+				},
+				null,   //userSelected
+				"27",   //androidApiLevel
+				"27",   //androidApiLevelName
+				"v8.1", //targetFrameworkVersion
+				"27",   //supportedApiLevel
+			},
+			//User selected a new API level 28 is not stable yet
+			new object [] {
+				"User Selected Unstable API", //description
+				// androidSdks
+				new [] {
+					new ApiInfo () { Id = "26", Level = 26, Name = "Oreo", FrameworkVersion = "v8.0", Stable = true },
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+					new ApiInfo () { Id = "28", Level = 28, Name = "P",    FrameworkVersion = "v9.0", Stable = false },
+				},
+				// targetFrameworks
+				new ApiInfo [] {
+					new ApiInfo () { Id = "26", Level = 26, Name = "Oreo", FrameworkVersion = "v8.0", Stable = true },
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+					new ApiInfo () { Id = "28", Level = 28, Name = "P",    FrameworkVersion = "v9.0", Stable = false },
+				},
+				"v9.0", //userSelected
+				"28",   //androidApiLevel
+				"28",   //androidApiLevelName
+				"v9.0", //targetFrameworkVersion
+				"28",   //supportedApiLevel
+			},
+		};
+
+		[Test]
+		[TestCaseSource (nameof (TargetFrameworkPairingParameters))]
+		public void TargetFrameworkPairing (string description, ApiInfo[] androidSdk, ApiInfo[] targetFrameworks, string userSelected, string androidApiLevel, string androidApiLevelName, string targetFrameworkVersion, string supportedApiLevel)
+		{
+			var path = Path.Combine ("temp", $"{nameof (TargetFrameworkPairing)}_{description}");
+			var androidSdkPath = CreateFauxAndroidSdkDirectory (Path.Combine (path, "android-sdk"), "26.0.3", androidSdk);
+			string javaExe = string.Empty;
+			string javacExe;
+			var javaPath = CreateFauxJavaSdkDirectory (Path.Combine (path, "jdk"), "1.8.0", out javaExe, out javacExe);
+			var referencePath = CreateFauxReferencesDirectory (Path.Combine (path, "references"), targetFrameworks);
+			IBuildEngine engine = new MockBuildEngine (TestContext.Out);
+			var resolveSdks = new ResolveSdks {
+				BuildEngine = engine,
+				AndroidSdkPath = androidSdkPath,
+				AndroidNdkPath = androidSdkPath,
+				JavaSdkPath = javaPath,
+				ReferenceAssemblyPaths = new [] {
+					Path.Combine (referencePath, "MonoAndroid"),
+				},
+			};
+			var androidTooling = new ResolveAndroidTooling {
+				BuildEngine = engine,
+				AndroidSdkPath = androidSdkPath,
+				AndroidNdkPath = androidSdkPath,
+				UseLatestAndroidPlatformSdk = true,
+				TargetFrameworkVersion = userSelected,
+			};
+			Assert.IsTrue (resolveSdks.Execute (), "ResolveSdks should succeed!");
+			Assert.IsTrue (androidTooling.Execute (), "ResolveAndroidTooling should succeed!");
+			Assert.AreEqual (androidApiLevel, androidTooling.AndroidApiLevel, $"AndroidApiLevel should be {androidApiLevel}");
+			Assert.AreEqual (androidApiLevelName, androidTooling.AndroidApiLevelName, $"AndroidApiLevelName should be {androidApiLevelName}");
+			Assert.AreEqual (targetFrameworkVersion, androidTooling.TargetFrameworkVersion, $"TargetFrameworkVersion should be {targetFrameworkVersion}");
+			Assert.AreEqual (supportedApiLevel, androidTooling.SupportedApiLevel, $"SupportedApiLevel should be {supportedApiLevel}");
 			Directory.Delete (Path.Combine (Root, path), recursive: true);
 		}
 	}

--- a/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/ResolveSdksTaskTests.cs
+++ b/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/ResolveSdksTaskTests.cs
@@ -250,7 +250,6 @@ namespace Xamarin.Android.Build.Tests {
 			Assert.AreEqual (androidTooling.AndroidApiLevel, "26", "AndroidApiLevel should be 26");
 			Assert.AreEqual (androidTooling.TargetFrameworkVersion, "v8.0", "TargetFrameworkVersion should be v8.0");
 			Assert.AreEqual (androidTooling.AndroidApiLevelName, "26", "AndroidApiLevelName should be 26");
-			Assert.AreEqual (androidTooling.SupportedApiLevel, "26", "SupportedApiLevel should be 26");
 			Assert.NotNull (resolveSdks.ReferenceAssemblyPaths, "ReferenceAssemblyPaths should not be null.");
 			Assert.AreEqual (resolveSdks.ReferenceAssemblyPaths.Length, 1, "ReferenceAssemblyPaths should have 1 entry.");
 			Assert.AreEqual (resolveSdks.ReferenceAssemblyPaths[0], Path.Combine (referencePath, "MonoAndroid"), $"ReferenceAssemblyPaths should be {Path.Combine (referencePath, "MonoAndroid")}.");
@@ -296,7 +295,6 @@ namespace Xamarin.Android.Build.Tests {
 				"27",   //androidApiLevel
 				"27",   //androidApiLevelName
 				"v8.1", //targetFrameworkVersion
-				"28",   //supportedApiLevel
 			},
 			//28 is installed but we only support 27
 			new object [] {
@@ -314,7 +312,6 @@ namespace Xamarin.Android.Build.Tests {
 				"27",   //androidApiLevel
 				"27",   //androidApiLevelName
 				"v8.1", //targetFrameworkVersion
-				"27",   //supportedApiLevel
 			},
 			//A paired downgrade to API 26
 			new object [] {
@@ -333,7 +330,6 @@ namespace Xamarin.Android.Build.Tests {
 				"26",   //androidApiLevel
 				"26",   //androidApiLevelName
 				"v8.0", //targetFrameworkVersion
-				"28",   //supportedApiLevel
 			},
 			//A new API level 28 is not stable yet
 			new object [] {
@@ -354,7 +350,6 @@ namespace Xamarin.Android.Build.Tests {
 				"27",   //androidApiLevel
 				"27",   //androidApiLevelName
 				"v8.1", //targetFrameworkVersion
-				"27",   //supportedApiLevel
 			},
 			//User selected a new API level 28 is not stable yet
 			new object [] {
@@ -375,13 +370,12 @@ namespace Xamarin.Android.Build.Tests {
 				"28",   //androidApiLevel
 				"28",   //androidApiLevelName
 				"v9.0", //targetFrameworkVersion
-				"28",   //supportedApiLevel
 			},
 		};
 
 		[Test]
 		[TestCaseSource (nameof (TargetFrameworkPairingParameters))]
-		public void TargetFrameworkPairing (string description, ApiInfo[] androidSdk, ApiInfo[] targetFrameworks, string userSelected, string androidApiLevel, string androidApiLevelName, string targetFrameworkVersion, string supportedApiLevel)
+		public void TargetFrameworkPairing (string description, ApiInfo[] androidSdk, ApiInfo[] targetFrameworks, string userSelected, string androidApiLevel, string androidApiLevelName, string targetFrameworkVersion)
 		{
 			var path = Path.Combine ("temp", $"{nameof (TargetFrameworkPairing)}_{description}");
 			var androidSdkPath = CreateFauxAndroidSdkDirectory (Path.Combine (path, "android-sdk"), "26.0.3", androidSdk);
@@ -411,7 +405,6 @@ namespace Xamarin.Android.Build.Tests {
 			Assert.AreEqual (androidApiLevel, androidTooling.AndroidApiLevel, $"AndroidApiLevel should be {androidApiLevel}");
 			Assert.AreEqual (androidApiLevelName, androidTooling.AndroidApiLevelName, $"AndroidApiLevelName should be {androidApiLevelName}");
 			Assert.AreEqual (targetFrameworkVersion, androidTooling.TargetFrameworkVersion, $"TargetFrameworkVersion should be {targetFrameworkVersion}");
-			Assert.AreEqual (supportedApiLevel, androidTooling.SupportedApiLevel, $"SupportedApiLevel should be {supportedApiLevel}");
 			Directory.Delete (Path.Combine (Root, path), recursive: true);
 		}
 	}

--- a/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets
+++ b/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets
@@ -726,7 +726,6 @@ Copyright (C) 2011-2012 Xamarin. All rights reserved.
 		<Output TaskParameter="TargetFrameworkVersion"      PropertyName="TargetFrameworkVersion" />
 		<Output TaskParameter="AndroidApiLevel"             PropertyName="_AndroidApiLevel"            Condition="'$(_AndroidApiLevel)' == ''" />
 		<Output TaskParameter="AndroidApiLevelName"         PropertyName="_AndroidApiLevelName" />
-		<Output TaskParameter="SupportedApiLevel"           PropertyName="_SupportedApiLevel" />
 		<Output TaskParameter="AndroidSdkBuildToolsPath"    PropertyName="AndroidSdkBuildToolsPath"    Condition="'$(AndroidSdkBuildToolsPath)' == ''" />
 		<Output TaskParameter="AndroidSdkBuildToolsBinPath" PropertyName="AndroidSdkBuildToolsBinPath" Condition="'$(AndroidSdkBuildToolsBinPath)' == ''" />
 		<Output TaskParameter="ZipAlignPath"                PropertyName="ZipAlignToolPath"            Condition="'$(ZipAlignToolPath)' == ''" />
@@ -1007,7 +1006,7 @@ because xbuild doesn't support framework reference assemblies.
 	</CreateProperty>
 
 	<!-- Get the defined constants for this API Level -->
-	<GetAndroidDefineConstants AndroidApiLevel="$(_SupportedApiLevel)" ProductVersion="$(MonoAndroidVersion)">
+	<GetAndroidDefineConstants AndroidApiLevel="$(_AndroidApiLevel)" ProductVersion="$(MonoAndroidVersion)">
 		<Output TaskParameter="AndroidDefineConstants" ItemName="AndroidDefineConstants" />
 	</GetAndroidDefineConstants>
 

